marine.c

```diff
@@ -564,7 +566,7 @@ int parse_output_fields(output_fields_t *output_fields, char **fields, int field
 WS_DLL_PUBLIC int validate_fields(char **fields, size_t fields_len) {
     output_fields_t *output_fields = output_fields_new();
 
-    if (parse_output_fields(output_fields, fields, fields_len, NULL) != 0) {
+    if (parse_output_fields(output_fields, fields, (int) fields_len, NULL) != 0) {
```

Comment:
I think the better thing to do is change the parameter type to `unsigned int`, since `fields_len` can't be negative.